modules/user/user.resolver.js

```diff
@@ -3,15 +3,19 @@ const userService = require('./user.service');
 const userQuery = {
   getAllUsers: (parent, args) => userService.getAllUsers(),
   getUserByToken: (parent, args, context) => (context.user
-    ? userService.getUserById(context.user)
+    ? userService.getUser(context.user.userId)
     : new Error('Unauthorized')),
+  getUserById: (parent, args, context) => (context.user ? userService.getUser(args.id) : new Error('Unauthorized')),
 };
 const userMutation = {
   registerUser: (parent, args) => userService.registerUser(args.user),
   loginUser: (parent, args) => userService.loginUser(args.user),
   deleteUser: (parent, args) => userService.deleteUser(args.id),
-  updateUser: (parent, args, context) => (context.user
-    ? userService.updateUser(args.user, context.user)
+  updateUserById: (parent, args, context) => (context.user
+    ? userService.updateUser(args.user, args.id)
+    : new Error('Unauthorized')),
+  updateUserByToken: (parent, args, context) => (context.user
+    ? userService.updateUser(args.user, context.user.userId)
```

Comment:
Try to implement separate method for updating user by token because in that method you don't need to check does user exist because it is already done inside the context builder

Reply:
fixed